src/player.cpp

```diff
@@ -2120,7 +2120,10 @@ void Player::despawn()
 		return;
 	}
 
-	sendCancelWalk();
+	if (!listWalkDir.empty()) {
```

Comment:
This does not change anything, the same situation continues to occur in which the character becomes immobile due to the desynchronization of the client with the server.

I just realized that you only need to add the function: `stopEventWalk();` below `sendCancelWalk();` and this question is solved.

the problem lies here:
```c++
if (eventWalk != 0) {
		return;
	}
```
By basic rules of death, this event must be executed again as if it were your first step, do not forget that even if the player is not eliminated, he is actually dying.

Reply:
https://github.com/otland/forgottenserver/blob/a540e72a10f68cbd0a2e04e489652db4f9ce7902/src/creature.cpp#L294

Reply:
> https://github.com/otland/forgottenserver/blob/a540e72a10f68cbd0a2e04e489652db4f9ce7902/src/creature.cpp#L292-L298
> 
> Honestly I have no idea what code snippet you posted.

sorry, in the function: `void Creature::addEventWalk(bool firstStep)`

Reply:
So what does addEventWalk has to do with the problem here?

Reply:
> So what does addEventWalk has to do with the problem here?

<details>
<summary>Image</summary>

![image](https://user-images.githubusercontent.com/28090948/104110246-49be6880-52ac-11eb-8a2f-ed97aaccdce5.png)

</details>

Reply:
Sorry, but I dont see the connection here. addEventWalk is called **each step** when player is auto walking. Player is auto walking basing on the listWalkDir container, so clearing listWalkDir and aborting the walk, cancels the auto walk.

https://github.com/otland/forgottenserver/blob/a540e72a10f68cbd0a2e04e489652db4f9ce7902/src/creature.cpp#L194

Reply:
:v for this:
![image](https://user-images.githubusercontent.com/28090948/104110549-14674a00-52af-11eb-9443-4c6fa866dc7d.png)

I'm not really sure, but apparently the reference is duplicated, or it may even lack it

Reply:
Yeah, calling stopEventWalk directly fixes it, but honestly I have no idea why xD

Reply:
> Yeah, calling stopEventWalk directly fixes it, but honestly I have no idea why xD
> Maybe the one additional eventWalk execution is breaking the client.
> 
> It's not redundant code, cause this method is only called when creature is following or auto walking. I bet it's kind of bug fix, but if you sure and you looked up all the traces, then you can push the change :D

You are right, this function is sometimes executed in the dispatcher thread, and it may happen that this list is filled with directions from the main thread or others...